File: app/src/main/java/md/fusionworks/lifehack/data/api/interceptor/HeaderInterceptor.java
package md.fusionworks.lifehack.data.api.interceptor;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * Created by ungvas on 1/23/16.
 */
public class HeaderInterceptor implements Interceptor {

  String apiKey;

  public HeaderInterceptor(String apiKey) {
    this.apiKey = apiKey;
  }

  @Override public Response intercept(Chain chain) throws IOException {
    Request original = chain.request();
    Request request = original.newBuilder()
        .header("Accept", "application/json")
        .header("authorization", apiKey)
        .build();
    return chain.proceed(request);
  }
}

File: app/src/main/java/md/fusionworks/lifehack/data/api/HeaderInterceptor.kt
package md.fusionworks.lifehack.data.api

import java.io.IOException

import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response

/**
 * Created by ungvas on 1/23/16.
 */
class HeaderInterceptor(internal var apiKey: String) : Interceptor {

  @Throws(IOException::class)
  override fun intercept(chain: Interceptor.Chain): Response {
    val original = chain.request()
    val request = original.newBuilder().header("Accept", "application/json").header("authorization",
        apiKey).build()
    return chain.proceed(request)
  }
}
